src/components/pagination/index.js

```diff
@@ -0,0 +1,183 @@
+        endPage = currentPage + 4
+      }
+    }
+
+    let startIndex = (currentPage - 1) * pageSize
+    let endIndex = Math.min(startIndex + pageSize - 1, totalItems - 1)
+
+    let pages = [...Array((endPage + 1) - startPage).keys()].map(i => startPage + i)
+
+    return {
+      totalItems: totalItems,
+      currentPage: currentPage,
+      pageSize: pageSize,
+      totalPages: totalPages,
+      startPage: startPage,
+      endPage: endPage,
+      startIndex: startIndex,
+      endIndex: endIndex,
+      pages: pages,
+    }
+  }
+
+  useEffect(() => {
+    setPage((items && items.length) && initialPage)
+  }, [items, rowsPerPageSize])
+
+  return (
+    <>
+      {pager.pages &&     
+      <ul className={`pagination ${paginationClasses.container}`}>
+        { rowsPerPage && 
+          <li className={'px-2'}>
+            <span className={'mx-2'}>rows per page: </span>
+            <select className={'rows-selection pr-5'} name="rowsPerPage" id="rowsPerPage" onChange={e => setRowsPerPageSize(parseInt(e.target.value))}>
+              {rowsPerPage.map((data, index) => {
+                return(
+                  <option key={index} value={data}>{data}</option>
+                )
+              })}
+              <option value={pager.totalItems}>All</option>
+            </select> 
+          </li>
+        }
+        { counter && 
+          <li className={'px-2'}>
+            <span>{pager.startIndex + 1}-{pager.endIndex + 1} of {pager.totalItems}</span>
+          </li>
+        }
+        { firstLast &&
+          <li className={`${paginationClasses.item} ${pager.currentPage === 1 ? 'disabled' : ''}`}>
+            <a onClick={() => setPage(1)}>First</a>
+          </li>
+        }
+        <li 
+          className={`
+            ${paginationClasses.item} 
+            ${pager.currentPage === 1 ? 'disabled' : ''}
+          `}
+        >
+          <a onClick={() => setPage(pager.currentPage - 1)}><ChevronLeft /></a>
+        </li>
+        {showPage && pager.pages.map((page, index) =>
+          <li 
+            key={index} 
+            className={`
+              ${paginationClasses.item}
+              ${paginationClasses.arrow}
+              ${paginationClasses.pageItem}
+              ${pager.currentPage === page ? paginationClasses.currentPageColor : ''}
+            `}
+          >
+            <a onClick={() => setPage(page)}>{page}</a>
+          </li>,
+        )}
+        <li 
+          className={`
+            ${paginationClasses.item}
+            ${paginationClasses.arrow} 
+            ${pager.currentPage === pager.totalPages ? 'disabled' : ''}
+          `}
+        >
+          <a onClick={() => setPage(pager.currentPage + 1)}><ChevronRight /></a>
+        </li>
+        {firstLast && 
+          <li className={`${paginationClasses.item} ${pager.currentPage === pager.totalPages ? 'disabled' : ''}`}>
+            <a onClick={() => setPage(pager.totalPages)}>Last</a>
+          </li>
+        }
```

Comment:
When `rowsPerPageSize` is the same as the total number of items, perhaps hide these?

Reply:
sounds good